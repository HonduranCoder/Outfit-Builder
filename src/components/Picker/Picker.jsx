import React from 'react';

export default function Picker({
  face,
  onFaceChange,
  top,
  onTopChange,
  bottom,
  onBottomChange,
  newCatchphrase,
  setCatchphrase,
  handleClick,
}) {
  const facePics = ['smiley', 'shy', 'observing'];
  const topPics = ['tank', 'jersey', 'polo'];
  const bottomPics = ['shorts', 'pants', 'skirt'];

  return (
    <div>
      <label>
        Face
        <select value={face} onChange={(e) => onFaceChange(e.target.value)}>
          {facePics.map((option) => (
            <option key={option}>{option}</option>
          ))}
        </select>
      </label>

      <label>
        Top
        <select value={top} onChange={(e) => onTopChange(e.target.value)}>
          {topPics.map((option) => (
            <option key={option}>{option}</option>
          ))}
        </select>
      </label>

      <label>
        Bottom
        <select value={bottom} onChange={(e) => onBottomChange(e.target.value)}>
          {bottomPics.map((option) => (
            <option key={option}>{option}</option>
          ))}
        </select>
      </label>

      <label>
        New Catchphrase:
        <input
          type="text"
          value={newCatchphrase}
          onChange={(e) => setCatchphrase(e.target.value)}
        />
      </label>
      <button type="button" onClick={handleClick}>
        Add Catchphrase!
      </button>
    </div>
  );
}
