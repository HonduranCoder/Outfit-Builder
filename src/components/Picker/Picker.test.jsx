import { screen, render } from '@testing-library/react';
import Picker from './Picker';

it('should render our Picker component', () => {
  const { container } = render(<Picker />);
  const selectFace = screen.getByLabelText('Face');
  const selectTop = screen.getByLabelText('Top');
  const selectBottom = screen.getByLabelText('Bottom');
  const input = screen.getByLabelText('New Catchphrase:');

  expect(selectFace).toBeInTheDocument();
  expect(selectTop).toBeInTheDocument();
  expect(selectBottom).toBeInTheDocument();
  expect(input).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
