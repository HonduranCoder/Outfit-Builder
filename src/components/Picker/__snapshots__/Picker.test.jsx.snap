// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render our Picker component 1`] = `
<div>
  <div>
    <label>
      Face
      <select>
        <option>
          smiley
        </option>
        <option>
          shy
        </option>
        <option>
          observing
        </option>
      </select>
    </label>
    <label>
      Top
      <select>
        <option>
          tank
        </option>
        <option>
          jersey
        </option>
        <option>
          polo
        </option>
      </select>
    </label>
    <label>
      Bottom
      <select>
        <option>
          shorts
        </option>
        <option>
          pants
        </option>
        <option>
          skirt
        </option>
      </select>
    </label>
    <label>
      New Catchphrase:
      <input
        type="text"
        value=""
      />
    </label>
    <button
      type="button"
    >
      Add Catchphrase!
    </button>
  </div>
</div>
`;
